### ISSUE TITLE
The training loss of the current version is very low

### ISSUE THREAD

**Issue by LoserCheems** [huggingface/trl#2875]
### Reproduction

https://github.com/huggingface/open-r1/blob/main/src/open_r1/sft.py

The training loss is very low, the same model is still above 1 in the previous version, and it is not good from the effect of checkpoints, repeating a sentence all the time.

{'loss': 5.9309, 'grad_norm': 15.3125, 'learning_rate': 5.509641873278237e-05, 'mean_token_accuracy': 0.1440385853435518, 'epoch': 0.03}                                              
  1%|█▉                                                                                                                                         | 100/7252 [24:16<29:02:42, 14.62s/it][INFO|trainer.py:3910] 2025-02-16 18:41:31,587 >> Saving model checkpoint to /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-100
[INFO|configuration_utils.py:420] 2025-02-16 18:41:31,589 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-100/config.json
[INFO|configuration_utils.py:909] 2025-02-16 18:41:31,590 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-100/generation_config.json
[INFO|modeling_utils.py:2988] 2025-02-16 18:41:32,073 >> Model weights saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-100/model.safetensors
[INFO|tokenization_utils_base.py:2491] 2025-02-16 18:41:32,076 >> tokenizer config file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-100/tokenizer_config.json
[INFO|tokenization_utils_base.py:2500] 2025-02-16 18:41:32,077 >> Special tokens file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-100/special_tokens_map.json
{'loss': 4.4443, 'grad_norm': 5.75, 'learning_rate': 0.00011019283746556474, 'mean_token_accuracy': 0.28513646389663827, 'epoch': 0.06}                                               
  3%|███▊                                                                                                                                       | 200/7252 [48:27<28:33:00, 14.57s/it][INFO|trainer.py:3910] 2025-02-16 19:05:42,751 >> Saving model checkpoint to /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-200
[INFO|configuration_utils.py:420] 2025-02-16 19:05:42,754 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-200/config.json
[INFO|configuration_utils.py:909] 2025-02-16 19:05:42,754 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-200/generation_config.json
[INFO|modeling_utils.py:2988] 2025-02-16 19:05:43,215 >> Model weights saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-200/model.safetensors
[INFO|tokenization_utils_base.py:2491] 2025-02-16 19:05:43,218 >> tokenizer config file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-200/tokenizer_config.json
[INFO|tokenization_utils_base.py:2500] 2025-02-16 19:05:43,219 >> Special tokens file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-200/special_tokens_map.json
{'loss': 0.1463, 'grad_norm': 0.154296875, 'learning_rate': 0.00016528925619834712, 'mean_token_accuracy': 0.9686542391721049, 'epoch': 0.08}                                         
  4%|█████▋                                                                                                                                   | 300/7252 [1:12:39<27:52:27, 14.43s/it][INFO|trainer.py:3910] 2025-02-16 19:29:54,473 >> Saving model checkpoint to /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-300
[INFO|configuration_utils.py:420] 2025-02-16 19:29:54,475 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-300/config.json
[INFO|configuration_utils.py:909] 2025-02-16 19:29:54,476 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-300/generation_config.json
[INFO|modeling_utils.py:2988] 2025-02-16 19:29:54,887 >> Model weights saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-300/model.safetensors
[INFO|tokenization_utils_base.py:2491] 2025-02-16 19:29:54,890 >> tokenizer config file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-300/tokenizer_config.json
[INFO|tokenization_utils_base.py:2500] 2025-02-16 19:29:54,890 >> Special tokens file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-300/special_tokens_map.json
{'loss': 0.0083, 'grad_norm': 0.12255859375, 'learning_rate': 0.00022038567493112948, 'mean_token_accuracy': 0.997627052143617, 'epoch': 0.11}                                        
  6%|███████▌                                                                                                                                 | 400/7252 [1:36:54<27:36:57, 14.51s/it][INFO|trainer.py:3910] 2025-02-16 19:54:09,855 >> Saving model checkpoint to /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-400
[INFO|configuration_utils.py:420] 2025-02-16 19:54:09,857 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-400/config.json
[INFO|configuration_utils.py:909] 2025-02-16 19:54:09,858 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-400/generation_config.json
[INFO|modeling_utils.py:2988] 2025-02-16 19:54:10,266 >> Model weights saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-400/model.safetensors
[INFO|tokenization_utils_base.py:2491] 2025-02-16 19:54:10,269 >> tokenizer config file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-400/tokenizer_config.json
[INFO|tokenization_utils_base.py:2500] 2025-02-16 19:54:10,269 >> Special tokens file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-400/special_tokens_map.json
{'loss': 0.0042, 'grad_norm': 0.0576171875, 'learning_rate': 0.0002754820936639118, 'mean_token_accuracy': 0.9985207893134777, 'epoch': 0.14}                                         
  7%|█████████▍                                                                                                                               | 500/7252 [2:01:16<27:19:14, 14.57s/it][INFO|trainer.py:3910] 2025-02-16 20:18:31,856 >> Saving model checkpoint to /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-500
[INFO|configuration_utils.py:420] 2025-02-16 20:18:31,858 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-500/config.json
[INFO|configuration_utils.py:909] 2025-02-16 20:18:31,858 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-500/generation_config.json
[INFO|modeling_utils.py:2988] 2025-02-16 20:18:32,291 >> Model weights saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-500/model.safetensors
[INFO|tokenization_utils_base.py:2491] 2025-02-16 20:18:32,293 >> tokenizer config file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-500/tokenizer_config.json
[INFO|tokenization_utils_base.py:2500] 2025-02-16 20:18:32,294 >> Special tokens file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-500/special_tokens_map.json
{'loss': 0.003, 'grad_norm': 0.025390625, 'learning_rate': 0.00033057851239669424, 'mean_token_accuracy': 0.9988212652492549, 'epoch': 0.17}                                          
  8%|███████████▎                                                                                                                             | 600/7252 [2:25:58<27:05:58, 14.67s/it][INFO|trainer.py:3910] 2025-02-16 20:43:13,243 >> Saving model checkpoint to /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-600
[INFO|configuration_utils.py:420] 2025-02-16 20:43:13,245 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-600/config.json
[INFO|configuration_utils.py:909] 2025-02-16 20:43:13,246 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-600/generation_config.json
[INFO|modeling_utils.py:2988] 2025-02-16 20:43:13,732 >> Model weights saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-600/model.safetensors
[INFO|tokenization_utils_base.py:2491] 2025-02-16 20:43:13,735 >> tokenizer config file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-600/tokenizer_config.json
[INFO|tokenization_utils_base.py:2500] 2025-02-16 20:43:13,735 >> Special tokens file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-600/special_tokens_map.json
{'loss': 0.0024, 'grad_norm': 0.047119140625, 'learning_rate': 0.0003856749311294766, 'mean_token_accuracy': 0.9990416939269987, 'epoch': 0.19}                                       
 10%|█████████████▏                                                                                                                           | 700/7252 [2:50:18<26:29:33, 14.56s/it][INFO|trainer.py:3910] 2025-02-16 21:07:33,305 >> Saving model checkpoint to /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-700
[INFO|configuration_utils.py:420] 2025-02-16 21:07:33,307 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-700/config.json
[INFO|configuration_utils.py:909] 2025-02-16 21:07:33,308 >> Configuration saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-700/generation_config.json
[INFO|modeling_utils.py:2988] 2025-02-16 21:07:33,739 >> Model weights saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-700/model.safetensors
[INFO|tokenization_utils_base.py:2491] 2025-02-16 21:07:33,741 >> tokenizer config file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-700/tokenizer_config.json
[INFO|tokenization_utils_base.py:2500] 2025-02-16 21:07:33,742 >> Special tokens file saved in /root/autodl-tmp/data/Doge-160M-Instruct-SFT/checkpoint-700/special_tokens_map.json

### System Info

Copy-paste the following information when reporting an issue:

- Platform: Linux-5.15.0-78-generic-x86_64-with-glibc2.35
- Python version: 3.12.3
- PyTorch version: 2.5.1+cu124
- CUDA device(s): NVIDIA GeForce RTX 4090
- Transformers version: 4.48.3
- Accelerate version: 1.3.0
- Accelerate config: not found
- Datasets version: 3.3.0
- HF Hub version: 0.28.1
- TRL version: 0.15.0
- bitsandbytes version: 0.45.2
- DeepSpeed version: not installed
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: not installed
- PEFT version: not installed

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete
